File: MessageList/app/src/main/java/ru/startandroid/messagelist/common/utils/DisposableManager.java
package ru.startandroid.messagelist.common.utils;

import io.reactivex.disposables.CompositeDisposable;
import io.reactivex.disposables.Disposable;

public class DisposableManager {

    private final CompositeDisposable compositeDisposable = new CompositeDisposable();

    public void add(Disposable disposable) {
        compositeDisposable.add(disposable);
    }

    public void remove(Disposable disposable) {
        if (disposable != null) {
            compositeDisposable.remove(disposable);
        }
    }

    public void clear() {
        compositeDisposable.clear();
    }

}

File: Vocabulary/app/src/main/java/ru/startandroid/vocabulary/common/utils/DisposableManager.kt
package ru.startandroid.vocabulary.common.utils

import io.reactivex.disposables.CompositeDisposable
import io.reactivex.disposables.Disposable

class DisposableManager {

    private val compositeDisposable = CompositeDisposable()

    fun add(disposable : Disposable) = compositeDisposable.add(disposable)

    fun remove(disposable : Disposable) = compositeDisposable.remove(disposable)

    fun clear() = compositeDisposable.clear();

}